relayer/src/modules/cosmos_to_eth.rs

```diff
@@ -0,0 +1,223 @@
+//! Defines Cosmos to Ethereum relayer module.
+
+use std::{net::SocketAddr, str::FromStr};
+
+use alloy::{
+    network::{Ethereum, EthereumWallet},
+    primitives::{Address, TxHash},
+    providers::{
+        fillers::{FillProvider, JoinFill, WalletFiller},
+        Identity, ProviderBuilder, RootProvider,
+    },
+    signers::local::PrivateKeySigner,
+    transports::BoxTransport,
+};
+use ibc_eureka_relayer_lib::{
+    listener::{cosmos_sdk, eth_eureka, ChainListenerService},
+    submitter::{
+        eth_eureka::{ChainSubmitter, SupportedProofType},
+        ChainSubmitterService,
+    },
+};
+use tendermint::Hash;
+use tendermint_rpc::{HttpClient, Url};
+use tonic::{transport::Server, Request, Response};
+
+use crate::{
+    api::{
+        self,
+        relayer_service_server::{RelayerService, RelayerServiceServer},
+    },
+    core::modules::{RelayerModule, RelayerModuleServer},
+};
+
+type Provider = FillProvider<
+    JoinFill<Identity, WalletFiller<EthereumWallet>>,
+    RootProvider<BoxTransport>,
+    BoxTransport,
+    Ethereum,
+>;
+
+/// The `RelayerModule` defines the relayer module for Cosmos to Ethereum.
+#[allow(clippy::module_name_repetitions)]
+pub struct CosmosToEthRelayerModule {
+    /// The chain listener for Cosmos SDK.
+    pub tm_listener: cosmos_sdk::ChainListener,
+    /// The chain listener for `EthEureka`.
+    pub eth_listener: eth_eureka::ChainListener<BoxTransport, Provider>,
+    /// The chain submitter for `EthEureka`.
+    pub submitter: ChainSubmitter<BoxTransport, Provider>,
+}
+
+/// The configuration for the Cosmos to Ethereum relayer module.
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
+#[allow(clippy::module_name_repetitions)]
+pub struct CosmosToEthConfig {
+    /// The tendermint RPC URL.
+    pub tm_rpc_url: String,
+    /// The ICS26 address.
+    pub ics26_address: Address,
+    /// The EVM RPC URL.
+    pub eth_rpc_url: String,
+    /// The private key for the Ethereum account.
+    // TODO: Use a more secure way to store the private key.
+    pub private_key: String,
+    /// The proof type to use for the SP1 ICS07 Tendermint prover.
+    /// This is either groth16 or plonk.
+    pub proof_type: String,
+    /// The SP1 prover network private key.
+    pub sp1_private_key: String,
+}
+
+#[tonic::async_trait]
+impl RelayerModule for CosmosToEthRelayerModule {
+    type Config = CosmosToEthConfig;
+
+    const NAME: &'static str = "cosmos_to_eth";
+
+    async fn new(config: Self::Config) -> Self {
+        let tm_client = HttpClient::new(
+            Url::from_str(&config.tm_rpc_url)
+                .unwrap_or_else(|_| panic!("invalid tendermint RPC URL: {}", config.tm_rpc_url)),
+        )
+        .expect("Failed to create tendermint HTTP client");
+
+        let tm_listener = cosmos_sdk::ChainListener::new(tm_client.clone());
+
+        let wallet = EthereumWallet::from(
```

Comment:
Since we don't need to sign anything, this seems unnecessary?

Reply:
Yes this is unnecessary. But I think nothing wrong with keeping it for now, I can remove it in the subsequent PR where we add actual relaying.